src/vs/workbench/common/events.ts

```diff
@@ -229,15 +234,15 @@ export class CommandEvent extends Event {
 }
 
 /**
- * Viewlet events are emitted when a viewlet opens or closes in the sidebar.
+ * Composite events are emitted when a composite opens or closes in the sidebar or panel.
  */
-export class ViewletEvent extends Event {
-	public viewletId: string;
+export class CompositeEvent extends Event {
+	public compositeId: string;
 
-	constructor(viewletId: string, originalEvent?: any) {
+	constructor(compositeId: string, originalEvent?: any) {
```

Comment:
Just tried this and the only issue is that sidebarPart.ts:90 sends a CompositeEvent before he has the actual composite, he just has the id. I can send the event later, on line 108 but not sure if too late